packages/importer/src/providers/standardnotes/index.ts

```diff
@@ -0,0 +1,248 @@
+          title: item.content.title,
+          dateCreated: item.created_at_timestamp,
+          dateEdited: item.updated_at_timestamp || item.created_at_timestamp,
+          pinned: item.content.appData["org.standardnotes.sn"]?.pinned,
+          tags: this.getTags(item, tags),
+          content: this.parseContent(item),
+        };
+        notes.push(note);
+      }
+
+      return Promise.resolve(true);
+    });
+  }
+
+  getContentType(item: SNBackupItem, components: SNBackupItem[]): EditorType {
+    let componentData =
+      item.content.appData["org.standardnotes.sn.components"] || {};
+    let editorId = Object.keys(componentData).pop();
+    if (editorId) {
+      let editor = components.find((component) => component.uuid === editorId);
+      if (editor) {
+        let contentType = {
+          ...editors[editor.content.name],
+        };
+        if (contentType.type === "code") {
+          //@ts-ignore
+          contentType.mode = componentData[editorId].mode;
+        }
+
+        return contentType;
+      }
+    }
+    return {
+      type:
+        item.content.text.includes("<") && item.content.text.includes("</")
+          ? "html"
+          : "text",
+    };
+  }
+
+  getTags(item: SNBackupItem, tags: SNBackupItem[]): string[] {
+    if (!item.content.references || item.content.references.length === 0)
+      return [];
+    let references = item.content.references;
+
+    let noteTags = [];
+    for (let reference of references) {
+      let tag = tags.find((tag) => tag.uuid === reference.uuid);
+      if (tag && tag.content.name && tag.content.name.trim() !== "") {
+        noteTags.push(tag.content.name);
+      }
+    }
+    return noteTags;
+  }
+
+  buildTableWithRows(rows: string) {
+    return `<div class="table-container">
+    <table>
+      <tbody>
+        ${rows}
+      </tbody>
+    </table>
+  </div>
+  <p><br data-mce-bogus="1"/></p>`;
+  }
+
+  /**
+   * Find object in array with maximum index
+   * @param array
+   * @returns
+   */
+  maxIndexItem(array: any[]) {
+    return array.reduce((prev, current) =>
+      prev.index > current.index ? prev : current
+    );
+  }
+
+  parseContent(item: SNBackupItem): Content {
+    const data = item.content.text;
+    const editorType = item.content.editorType;
+    switch (editorType.type) {
+      case "text":
+        return {
+          data: converter.makeHtml(data),
+          type: ContentType.HTML,
+        };
+      case "html":
+        return {
+          data: data,
+          type: ContentType.HTML,
+        };
+      case "markdown":
+        return {
+          data: converter.makeHtml(data),
+          type: ContentType.HTML,
+        };
+      case "code":
+        let language = editorType.mode || "plaintext";
+        if (language === "htmlmixed") language = "html";
+        let code = hljs.highlightAuto(data, [language]);
+        let html = `<pre class="hljs language-${language}">
+          ${code.value}
+        </pre>`;
```

Comment:
We should refactor the html construction out of the main logic ‒ one mistake here and it'll break the whole note.